### DISCUSSION TITLE
Is there a way to duplicate a buffer view by dragging it while holding some specific keys? (like cmd+option+drag in VSCode on macOS)

### DISCUSSION THREAD

**Discussion by umajho** [zed-industries/zed#14858] [Feature Requests]
This is how it work in VSCode (holding cmd+option while dragging):
![2024-07-20 17 50 19](https://github.com/user-attachments/assets/0bcfe07d-2b6d-484c-8cf8-5e8201a4071b)

In Zed, with `"base_keymap": "VSCode"`, holding those keys while dragging doesn't affect Zed's behavior: it just move the buffer being dragged.

**Comment by SomeoneToIgnore** [comment]
It's called splitting and works already in Zed without any extra keys added, just by dragging files — you'll see a similar highlight when it happens.

The only culprit is that __the only__ file will move instead of creating new splits.
For mac, you can use `cmd-N` or any of the keymaps below (check the default keymaps for Linux for similar thing) :

```jsonc

"cmd-\\": "pane::SplitRight",
{
    "context": "Pane",
    "bindings": {
      "cmd-k up": "pane::SplitUp",
      "cmd-k down": "pane::SplitDown",
      "cmd-k left": "pane::SplitLeft",
      "cmd-k right": "pane::SplitRight"
    }
  },
```

**Reply by umajho** [reply]
I'm sorry I didn't make it clear. I specifically mean using drag to duplicate. When the editor already has two panes side by side, if you want to display the file on the left side in both panes simultaneously, you can do it with a single drag if the feature I demonstrated in VSCode is available. Otherwise, you would have to split first and then drag, which adds an extra step.

**Reply by SomeoneToIgnore** [reply]
Yeah, it makes sense totally, I would suggest filing another issue (or, maybe there's one already?) for this.

**Reply by umajho** [reply]
I searched for "drag" + "split" and didn't find any existing issues, so I opened a new one (#14861). Thank you for your help.
